nym-vpn-apple/Services/Sources/Services/ConnectionManager/ConnectionManager.swift

```diff
@@ -287,13 +287,24 @@ private extension ConnectionManager {
 
     func disconnectActiveTunnel() {
         guard let activeTunnel,
-              activeTunnel.status == .connected || activeTunnel.status == .connecting
+              shouldDisconnectActiveTunnel()
         else {
             return
         }
         tunnelsManager.disconnect(tunnel: activeTunnel)
     }
 
+    func shouldDisconnectActiveTunnel() -> Bool {
+        guard let activeTunnel,
```

Comment:
Why so complicated? It hurts readability, could switch covering all enum variants:

```swift
switch activeTunnel.status {
  case .connected, .connecting, .reasserting: true
  case .invalid, disconnecting, .disconnected: false
}
```